### DISCUSSION TITLE
FluentDataGrid : what is the recommended way to apply style on FluentDataGrid, TemplateColumn, PropertyColumn

### DISCUSSION THREAD

**Discussion by jobou363** [microsoft/fluentui-blazor#1434] [General]
I'm using version 4.3.1 with .net 8.

I want to set distinct size on my FluentDataGrid columns but i find it hard to do.

I have tried to use class without success. 

I have succeed using style width but the data grid look not good (the bar is not continuing so the next look outbound) for the third column there is a visual issue i'm not sure how to fix it.
 
![image](https://github.com/microsoft/fluentui-blazor/assets/240580/6bbe2c44-1d80-43b3-9fc7-bd308bd196f5)

` <FluentDataGrid Class="mt-3" Context="GridContext" ItemsProvider="GridItemsProvider" TGridItem="Dtos" GenerateHeader="GenerateHeaderOption.Default">

   <TemplateColumn Align="@Align.Center" Style="width: 130px">
    <FluentButton IconStart="@(new Icons.Regular.Size20.Link())" OnClick="@(() => NavigateToWithParameter(TaskAction, context.GroupeEntityId))" >
       @TaskAction?.ActionProperties.GetDescription()
    </FluentButton>
</TemplateColumn>

     <PropertyColumn Class="width-100px" Style="width: 150px" Title="@Localization.Abreviation" Property="@(p => p.Abbreviation)" Sortable="true" />
     <PropertyColumn Title="@Localization.Nom" Style="width: 600px" Property="@(p => p.Nom)" Sortable="true" />

 </FluentDataGrid>`

**Comment by jobou363** [comment]
looks like i just add to but a Style for the total of my column so the header will fit the content = "width: 880px"

**Comment by vnbaaij** [comment]
You could also use the GridTemplateColumns parameter on the grid itself to specify the column widths

**Reply by jobou363** [reply]
Thank you, I prefer this GridTemplateColumns solution of one setting instead of 3 css style attributes to manage and really simple to use once you understand the ratio to use.

GridTemplateColumns="0.5fr 0.5fr 1fr"
